src/pages/lesson-page/lesson-page.tsx

```diff
@@ -1,28 +1,105 @@
-import classNames from 'classnames';
-import styles from './lesson-page.module.scss';
+import { ChevronLeftIcon } from '@radix-ui/react-icons';
 import commonStyles from '@styles/common-styles.module.scss';
+import classNames from 'classnames';
+import { format } from 'date-fns';
+import { useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { RouteParams } from '/src/router/config';
+import CommonStyles_module from '../../styles/common-styles.module.scss';
+import styles from './lesson-page.module.scss';
 import { useAvailability, useLessonBySlug } from '/src/api/api-hooks';
-import { useState } from 'react';
 import { Calendar } from '/src/components/calendar/calendar';
+import { LessonDetails } from '/src/components/lesson-details/lesson-details';
+import { RouteParams } from '/src/router/config';
 
-export interface LessonPageProps {
-    className?: string;
-}
-
-export const LessonPage = ({ className }: LessonPageProps) => {
+export const LessonPage: React.FC = () => {
     const { slug } = useParams<RouteParams['/lesson/:slug']>();
     const { data } = useLessonBySlug(slug);
     const { data: availability, isLoading } = useAvailability(data?._id!);
-    const [selectedDate, setSelectedDate] = useState(new Date());
 
-    if (!availability && isLoading) {
+    const [selectedDate, onDateSlected] = useState<Date>(new Date());
+    const [selectedHour, setSelectedHour] = useState<string>('');
+    const normalizeDate = (date: Date): string => {
+        return format(date, 'yyyy-MM-dd');
+    };
+
+    const typeOfClass = data?.name!;
+    const normalizedDatesSet = new Set<string>();
+    const lessonsByDate: Map<string, string[]> = new Map<string, string[]>();
+
+    const availableDates = availability?.availabilityEntries.reduce(
+        (acc, entry) => {
+            const currentDay: Date = new Date(entry?.slot?.startDate!);
+            const normalizedDay: string = normalizeDate(currentDay);
+            const startHour: string = format(currentDay, 'HH:mm:aa');
+
+            if (!normalizedDatesSet.has(normalizedDay)) {
+                acc.availableDates.push(currentDay);
```

Comment:
acc can be a set and then you don't need `normalizedDatesSet`

Reply:
or, actually you need only the map

Reply:
We can't use Date objects directly as keys in the Map because they can cause issues with key comparisons, especially when displaying buttons later on.